core/src/sumeragi/main_loop.rs

```diff
@@ -650,8 +651,14 @@ impl Sumeragi {
                         .chain(current_view_change_index, &mut new_wsv)
                         .sign(&self.key_pair);
 
+                        let created_in = create_block_start_time.elapsed();
                         if let Some(current_topology) = current_topology.is_consensus_required() {
-                            info!(%addr, block_payload_hash=%new_block.payload().hash(), "Block created");
+                            info!(%addr, created_in_ms=%created_in.as_millis(), block_payload_hash=%new_block.payload().hash(), "Block created");
+
+                            if created_in > self.pipeline_time() / 2 {
```

Comment:
what would be appropriate here?

Reply:
Half I think is good. Full block will get committed as fast as possible anyways.